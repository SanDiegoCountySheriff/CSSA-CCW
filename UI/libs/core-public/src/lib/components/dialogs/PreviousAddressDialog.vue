<template>
  <div>
    <v-dialog v-model="dialog">
      <template #activator="{ on, attrs }">
        <v-btn
          color="primary my-5"
          v-bind="attrs"
          v-on="on"
        >
          {{ $t(' Add previous address') }}
        </v-btn>
      </template>
      <div class="address-container">
        <v-form
          ref="form"
          v-model="valid"
        >
          <v-row>
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="address.addressLine1"
                label="Address line 1"
                :rules="[v => !!v || 'Address line 1 cannot be blank']"
              />
            </v-col>

            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="address.addressLine2"
                label="Address line 2"
              />
            </v-col>
          </v-row>

          <v-row>
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="address.city"
                label="City"
                :rules="[v => !!v || 'City cannot be blank']"
              />
            </v-col>

            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="address.state"
                label="State"
                :rules="[v => !!v || 'State cannot be blank&quot']"
              />
            </v-col>
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="address.county"
                label="County"
                :rules="[v => !!v || 'County cannot be blank']"
              />
            </v-col>
            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="address.zip"
                label="Zip"
                :rules="[v => !!v || 'Zip cannot be blank']"
              />
            </v-col>

            <v-col
              cols="6"
              md="5"
              sm="3"
            >
              <v-text-field
                v-model="address.country"
                label="Country"
                :rules="[v => !!v || 'Country cannot be blank']"
              />
            </v-col>
          </v-row>
        </v-form>
        <div class="mt-2 btn-container">
          <v-btn
            color="success"
            @click="handleSubmit"
            class="mr-2"
          >
            {{ $t('Submit') }}
          </v-btn>
          <v-btn
            color="error"
            @click="dialog = false"
          >
            {{ $t('Close') }}
          </v-btn>
        </div>
      </div>
    </v-dialog>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from "vue";
import { AddressInfo } from "@shared-ui/types/defualtTypes";

export default defineComponent({
  name: 'PreviousAddressDialog',
  props: {
    getPreviousAddressFromDialog: {
      type: Function as PropType<(address: AddressInfo) => void>,
      default: () => null,
    },
  },
  data() {
    return {
      address: {} as AddressInfo,
      dialog: false,
      valid: false,
    };
  },
  methods: {
    handleSubmit() {
      this.getPreviousAddressFromDialog(this.address);
    },
  },
});
</script>

<style lang="scss" scoped>
.address-container {
  display: flex;
  flex-direction: column;
  height: 50vh;
  width: 90%;
  justify-content: center;
  align-items: center;
  background: aliceblue;
  border-radius: 12px;
}
.btn-container {
  display: flex;
  width: 75%;
  justify-content: flex-end;
}
</style>
