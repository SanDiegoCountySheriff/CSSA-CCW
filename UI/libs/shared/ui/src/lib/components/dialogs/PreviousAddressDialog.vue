<template>
  <v-dialog
    v-model="dialog"
    max-width="800"
  >
    <template #activator="{ on, attrs }">
      <v-btn
        :disabled="readonly"
        small
        color="primary"
        v-bind="attrs"
        v-on="on"
      >
        {{ $t(' Add Previous Address') }}
      </v-btn>
    </template>

    <v-card outlined>
      <v-card-title>{{ $t('Previous Address Information') }}</v-card-title>

      <v-card-text>
        <v-form
          ref="form"
          v-model="valid"
        >
          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="$vuetify.breakpoint.smAndDown ? 'pb-0' : ''"
            >
              <v-text-field
                maxlength="150"
                v-model="state.address.streetAddress"
                :label="$t('Street Address')"
                :rules="[v => !!v || 'Street address cannot be blank']"
                outlined
                dense
              />
            </v-col>

            <v-col
              md="6"
              cols="12"
              :class="$vuetify.breakpoint.smAndDown ? 'pb-0' : ''"
            >
              <v-text-field
                maxlength="100"
                v-model="state.address.city"
                :label="$t('City')"
                :rules="[v => !!v || 'City cannot be blank']"
                outlined
                dense
              />
            </v-col>
          </v-row>

          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="$vuetify.breakpoint.smAndDown ? 'pb-0' : ''"
            >
              <v-combobox
                v-if="state.address.country === 'United States'"
                maxlength="100"
                :items="states"
                v-model="state.address.state"
                :label="$t('State')"
                :rules="[v => !!v || 'State cannot be blank']"
                outlined
                dense
              />

              <v-text-field
                v-if="state.address.country !== 'United States'"
                :label="$t('State / Region')"
                :rules="[v => !!v || $t('State / Region cannot be blank')]"
                v-model="state.address.state"
                outlined
                dense
              />
            </v-col>

            <v-col
              md="6"
              cols="12"
              :class="$vuetify.breakpoint.smAndDown ? 'pb-0' : ''"
            >
              <v-text-field
                maxlength="100"
                persistent-hint
                :hint="$t('If not applicable enter N/A ')"
                v-model="state.address.county"
                :label="$t('County')"
                :rules="[v => !!v || 'County cannot be blank']"
                outlined
                dense
              />
            </v-col>
          </v-row>

          <v-row>
            <v-col
              md="6"
              cols="12"
              :class="$vuetify.breakpoint.smAndDown ? 'pb-0' : ''"
            >
              <v-text-field
                maxlength="10"
                required
                persistent-hint
                v-model="state.address.zip"
                :label="$t('Zip')"
                :hint="$t('If not applicable enter N/A ')"
                :rules="[
                  v => !!v || $t('Field is required'),
                  v =>
                    !!v.match(/^\d+$/) ||
                    v === 'N/A' ||
                    v === 'n/a' ||
                    $t('Must contain only numbers'),
                ]"
                outlined
                dense
              />
            </v-col>

            <v-col
              md="6"
              cols="12"
              :class="$vuetify.breakpoint.smAndDown ? 'pb-0' : ''"
            >
              <v-combobox
                :items="countries"
                v-model="state.address.country"
                :label="$t('Country')"
                :rules="[v => !!v || 'Country cannot be blank']"
                outlined
                dense
              />
            </v-col>
          </v-row>
        </v-form>
      </v-card-text>

      <v-card-actions>
        <v-btn
          text
          color="error"
          @click="dialog = false"
        >
          {{ $t('Cancel') }}
        </v-btn>

        <v-spacer></v-spacer>

        <v-btn
          text
          color="primary"
          @click="handleSubmit"
          :disabled="!valid"
        >
          {{ $t('Submit') }}
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script setup lang="ts">
import { AddressInfoType } from '@shared-utils/types/defaultTypes'
import { countries, states } from '@shared-utils/lists/defaultConstants'
import { inject, reactive, ref } from 'vue'

const emit = defineEmits(['get-previous-address-from-dialog'])

const readonly = inject<boolean>('readonly', false)

const state = reactive({
  address: {
    streetAddress: '',
    city: '',
    country: '',
    county: '',
    state: '',
    zip: '',
  } as AddressInfoType,
})

let dialog = ref(false)
const valid = ref(false)

function handleSubmit() {
  emit('get-previous-address-from-dialog', state.address)

  state.address = reactive({
    streetAddress: '',
    city: '',
    country: '',
    county: '',
    state: '',
    zip: '',
  } as AddressInfoType)

  dialog.value = false
}
</script>
