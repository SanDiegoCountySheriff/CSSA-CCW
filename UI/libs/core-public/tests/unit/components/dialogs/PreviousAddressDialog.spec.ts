/* eslint-disable @typescript-eslint/ban-ts-comment */
import PreviousAddressDialog from '@shared-ui/components/dialogs/PreviousAddressDialog.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('PreviousAddressDialog', () => {
  let vuetify;
  let wrapper;

  beforeEach(async () => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(PreviousAddressDialog, {
      localVue,
      vuetify,
      mocks: tMock,
      data() {
        return { dialog: true };
      },
    });
    await wrapper.find('#add-previous-address-btn').trigger('click');
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the add button', async () => {
    expect(wrapper.find('button').exists()).toBeTruthy();
    expect(wrapper.find('form').exists()).toBeTruthy();
  });

  it('Should render the form', () => {
    expect(wrapper.find('form').exists()).toBeTruthy();
  });

  it('Should not allow submission', () => {
    expect(
      wrapper.find('#pre-address-submit-btn').attributes().disabled
    ).toBeTruthy();
  });
});
