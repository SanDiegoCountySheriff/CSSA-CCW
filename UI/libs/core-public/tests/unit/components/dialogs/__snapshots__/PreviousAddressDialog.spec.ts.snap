// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousAddressDialog should match the snapshot 1`] = `
<div><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary" id="add-previous-address-btn" role="button" aria-haspopup="true" aria-expanded="true"><span class="v-btn__content">
         Add previous address
      </span></button>
  <div class="v-dialog__container">
    <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
      <transition-stub name="dialog-transition" appear="true">
        <div tabindex="0" class="v-dialog v-dialog--active" style="transform-origin: center center;">
          <div class="address-container" style="background: rgb(238, 238, 238);">
            <form novalidate="novalidate" class="v-form form-container">
              <div class="row">
                <div class="col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="address-line-1" class="v-label theme--light" style="left: 0px; position: absolute;">Address line 1</label><input id="address-line-1" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-lg-6 col-12">
                  <div class="v-input pl-6 v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="input-12" class="v-label theme--light" style="left: 0px; position: absolute;">Address line 2</label><input id="input-12" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="row">
                <div class="col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="city" class="v-label theme--light" style="left: 0px; position: absolute;">City</label><input id="city" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-19" class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-select__slot"><label for="state" class="v-label theme--light" style="left: 0px; position: absolute;">State</label><input id="state" type="text" autocomplete="off">
                          <div class="v-input__append-inner">
                            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                          </div><input type="hidden">
                        </div>
                        <div class="v-menu">
                          <!---->
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="county" class="v-label theme--light" style="left: 0px; position: absolute;">County</label><input id="county" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-text-field__slot"><label for="zip" class="v-label theme--light" style="left: 0px; position: absolute;">Zip</label><input id="zip" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="col-lg-6 col-12">
                  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
                    <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star theme--light error--text" style="font-size: 12px;"></i></div>
                    <div class="v-input__control">
                      <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-33" class="v-input__slot">
                        <fieldset aria-hidden="true">
                          <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                        </fieldset>
                        <div class="v-select__slot"><label for="country" class="v-label theme--light" style="left: 0px; position: absolute;">Country</label><input id="country" type="text" autocomplete="off">
                          <div class="v-input__append-inner">
                            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                          </div><input type="hidden">
                        </div>
                        <div class="v-menu">
                          <!---->
                        </div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
            <div class="mt-2 btn-container"><button type="button" disabled="disabled" class="mr-2 v-btn v-btn--disabled v-btn--has-bg theme--light v-size--small" id="pre-address-submit-btn"><span class="v-btn__content">
          Submit
        </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small error"><span class="v-btn__content">
          Close
        </span></button></div>
          </div>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;
